### PR TITLE
Recreates article and definition

### PR DESCRIPTION
**Pull request by Rafase282**
Relates to #651

### CODE REVIEW

File: Online-VS-Offline-IDE.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Online vs Offline IDE
+
+## Offline IDEs
+
+There are several offline [IDE](Integrated Development Environment) that programmers use in order to stay more productive to learn and be more productive at work.
```
Comment (theflametrooper) [inline Online-VS-Offline-IDE.md]: Maybe you need to refer IDE as **plural**.

File: Online-VS-Offline-IDE.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+- **[Eclipse](https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html)** is more than a Java editor. Its biggest advantages are code completion by tabbing method which saves a lot of time writing documentations.
```
Comment (theflametrooper) [inline Online-VS-Offline-IDE.md]: documentations should just be **documentation**

File: Online-VS-Offline-IDE.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  It has a build in syntax check to fix any mistype words while coding a project. All of features we've come to expect of an IDE are their code completion, templates, integration  with various SCMSs, and integration with build systems.
+
+  Its code formatting and cleanup tools are very powerful. I find its build system to work well and intuitively. I think these are the things upon which its reputation is really built.
+
+  Furthermore it has a refactoring that find and replace function, variables and classes of a specific work.
+
+  Eclipse offers more plugins, platforms, frameworks, and simpler to use compare  to other IDE such as **IntelliJ**, **BlueJ** and **JEdit**.
```
Comment (theflametrooper) [inline Online-VS-Offline-IDE.md]: and **is** simpler to use **compared**

IDE should also be **plural**

File: Online-VS-Offline-IDE.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+- **[NetBeans](https://netbeans.org/features/)** is a free intuitive editor that does it all compared from other IDE. It has simple Swing GUI design tool to build user interfaces by dragging and dropping components for instance buttons, textboxes.
```
Comment (theflametrooper) [inline Online-VS-Offline-IDE.md]: IDE should be **plural**

File: Online-VS-Offline-IDE.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+## Conclusion
+
+There are several differences between online editors and the offline editors.
+
+An offline IDE will be more beneficial to a programmer rather than an online IDE due to the fact that large projects will slow down, and a software team will need a reasonable office internet service.
+
+If there is a major storm the connection will be affected causing many projects to be unfinished and incomplete.
+
+The only advantages of online IDE is that projects can be access everywhere; there are no setup and configurations.  There is no hardware limitation since all information will be stores in cloud.
```
Comment (theflametrooper) [inline Online-VS-Offline-IDE.md]: **advantage** of **an** online IDE

File: Online-VS-Offline-IDE.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+  The _disadvantage_ it is that with all its features it is slow to load and uses  more memory than other IDE's.
+
+- **[IntelliJ IDEA](https://www.jetbrains.com/idea/)** it is the most efficient offline editor however it is not free. It is faster than most editors however it has numbers of problems. Some major advantages is using too much memory from system,
+- **[BlueJ](http://www.bluej.org/)** is a simple editor mostly used to teach java and object oriented programming. The main advantage of the editor over other ones is that it does not need a main method to run the program.
+
+## Online IDEs
+
+- **[CodeRun Studio](http://www.webappers.com/2009/12/04/coderun-studio-free-cross-platform-online-ide/)** is an online IDE that you build web application. It allows you to run and test code online.
+
```
Comment (alayek) [inline Online-VS-Offline-IDE.md]: Do we want to mentioon c9.io or nitrous.io here?

File: Online-VS-Offline-IDE.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+- **[Eclipse](https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html)** is more than a Java editor. Its biggest advantages are code completion by tabbing method which saves a lot of time when writing documentation.
+
+  It has a build in syntax check to fix any mistype words while coding a project. All of features we've come to expect of an IDE are their code completion, templates, integration  with various SCMSs, and integration with build systems.
+
+  Its code formatting and cleanup tools are very powerful. I find its build system to work well and intuitively. I think these are the things upon which its reputation is really built.
+
+  Furthermore it has a refactoring that find and replace function, variables and classes of a specific work.
+
+  Eclipse offers more plugins, platforms, frameworks, and is simpler to use compare  to other IDEs such as **IntelliJ**, **BlueJ** and **JEdit**.
```
Comment (alayek) [inline Online-VS-Offline-IDE.md]: BlueJ is the simplest of IDEs and used to taught Java to teenage kids. Also, I don't think Eclipse is simpler or offers more than IntelliJ Idea.

File: Online-VS-Offline-IDE.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+# Online vs Offline IDEs
+
+## Offline IDEs
+
+There are several offline [IDE](Integrated Development Environment) that programmers use in order to stay more productive to learn and be more productive at work.
+
+- **[Eclipse](https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html)** is more than a Java editor. Its biggest advantages are code completion by tabbing method which saves a lot of time when writing documentation.
+
+  It has a build in syntax check to fix any mistype words while coding a project. All of features we've come to expect of an IDE are their code completion, templates, integration  with various SCMSs, and integration with build systems.
```
Comment (alayek) [inline Online-VS-Offline-IDE.md]: > ~~build in~~built-in syntax check

File: Online-VS-Offline-IDE.md (hunk 6 above)
Comment (Rafase282) [inline Online-VS-Offline-IDE.md]: We could, I didn't change much from the original as I wanted him to author it. I personally never heard of nitrous.io

**Pull request merged**